### PR TITLE
[SUB-TASK][KPIP-4] Enhance the response error msg

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

closes https://github.com/apache/incubator-kyuubi/issues/2632

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkBatchProcessBuilder.scala [hunk 1]
```diff
@@ -36,11 +36,13 @@ class SparkBatchProcessBuilder(
 
   override def mainResource: Option[String] = Option(batchRequest.resource)
 
-  override protected def commands: Array[String] = {
+  override protected val commands: Array[String] = {
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkBatchProcessBuilder.scala:39]: the commands will be several times, e.g. `builder.toString`

**Comment by ulysses-you** [comment]
cc @turboFei @zhaomin1423

**Review by turboFei — APPROVED** [review]
LGTM

**Review by zhaomin1423 — APPROVED** [review]
LGTM

**Comment by turboFei** [comment]
thanks, merging to master

**Comment by ulysses-you** [comment]
thank you @turboFei @zhaomin1423  for review
